modules/fluid_properties/src/userobjects/CO2FluidProperties.C

```diff
@@ -0,0 +1,720 @@
+  return phi0 + phir;
+}
+
+Real
+CO2FluidProperties::dphiSW_dd(Real delta, Real tau) const
+{
+  // Derivative of the ideal gas component wrt gamma
+  Real dphi0dd = 1.0 / delta;
+
+  // Derivative of the residual component wrt gamma
+  Real theta, Delta, Psi, dDelta_dd, dPsi_dd;
+  Real dphirdd = 0.0;
+
+  for (unsigned int i = 0; i < _n1.size(); ++i)
+    dphirdd += _n1[i] * _d1[i] * std::pow(delta, _d1[i] - 1.0) * std::pow(tau, _t1[i]);
+
+  for (unsigned int i = 0; i < _n2.size(); ++i)
+    dphirdd += _n2[i] * std::exp(- std::pow(delta, _c2[i])) * (std::pow(delta, _d2[i] - 1.0)
+      * std::pow(tau, _t2[i]) * (_d2[i] - _c2[i] * std::pow(delta, _c2[i])));
+
+  for (unsigned int i = 0; i < _n3.size(); ++i)
+    dphirdd += _n3[i] * std::pow(delta, _d3[i]) * std::pow(tau, _t3[i])
+      * std::exp(- _alpha3[i] * std::pow(delta - _eps3[i], 2.0) - _beta3[i] * std::pow(tau - _gamma3[i], 2.0))
+      * (_d3[i] / delta - 2.0 * _alpha3[i] * (delta - _eps3[i]));
+
+  for (unsigned int i = 0; i < _n4.size(); ++i)
+  {
+    theta = 1.0 - tau + _A4[i] * std::pow(std::pow(delta - 1.0, 2.0), 1.0 / (2.0 * _beta4[i]));
+    Delta = std::pow(theta, 2.0) + _B4[i] * std::pow(std::pow(delta - 1.0, 2.0), _a4[i]);
+    Psi = std::exp(- _C4[i] * std::pow(delta - 1.0, 2.0) - _D4[i] * std::pow(tau - 1.0, 2.0));
+    dPsi_dd = - 2.0 * _C4[i] * (delta - 1.0) * Psi;
+    dDelta_dd = (delta - 1.0) * (_A4[i] * theta * 2.0 / _beta4[i] *
+      std::pow(std::pow(delta - 1.0, 2.0), 1.0 / (2.0 * _beta4[i]) - 1.0) + 2.0 * _B4[i] * _a4[i]
+      * std::pow(std::pow(delta - 1.0, 2.0), _a4[i] - 1.0));
+
+    dphirdd += _n4[i] * (std::pow(Delta, _b4[i]) * (Psi + delta * dPsi_dd) + _b4[i] *
+      std::pow(Delta, _b4[i] - 1.0) * dDelta_dd * delta * Psi);
+  }
+
+  // The derivative of the free energy wrt delta is the sum of these components
+  return dphi0dd + dphirdd;
+}
+
+Real
+CO2FluidProperties::dphiSW_dt(Real delta, Real tau) const
+{
+  // Derivative of the ideal gas component wrt tau
+  Real sum0 = 0.0;
+  for (unsigned int i = 0; i < _a0.size(); ++i)
+    sum0 += _a0[i] * _theta0[i] * (1.0 / (1.0 - std::exp(- _theta0[i] * tau)) - 1.0);
+
+  Real dphi0dt = - 3.70454304 + 2.5 / tau + sum0;
+
+  // Derivative of the residual component wrt tau
+  Real theta, Delta, Psi, dDelta_dt, dPsi_dt;
+  Real dphirdt = 0.0;
+  for (unsigned int i = 0; i < _n1.size(); ++i)
+    dphirdt += _n1[i] * _t1[i] * std::pow(delta, _d1[i]) * std::pow(tau, _t1[i] - 1.0);
+
+  for (unsigned int i = 0; i < _n2.size(); ++i)
+    dphirdt += _n2[i] * _t2[i] * std::pow(delta, _d2[i]) * std::pow(tau, _t2[i] - 1.0) *
+      std::exp(- std::pow(delta, _c2[i]));
+
+  for (unsigned int i = 0; i < _n3.size(); ++i)
+    dphirdt += _n3[i] * std::pow(delta, _d3[i]) * std::pow(tau, _t3[i]) * std::exp(- _alpha3[i]
+      * std::pow(delta - _eps3[i], 2.0) - _beta3[i] * std::pow(tau - _gamma3[i], 2.0)) * (_t3[i] / tau
+      - 2.0 * _beta3[i] * (tau - _gamma3[i]));
+
+ for (unsigned int i = 0; i < _n4.size(); ++i)
```

Comment:
I've fixed this one, and also added a `mooseError` in beta as it isn't implemented. Thanks again for looking at these